suite/tests/client-interface/features.dll.c

```diff
@@ -71,12 +73,36 @@ read_hw_features(char *feat_str)
     return 1;
 }
 
+/*
+ * FEATURE_PAUTH is identified by six different nibbles. We initially check the API nibble
+ * but on Neoverse V1 hardware this is 0. Running this on a Neoverse V1 machine verifies
+ * that the other nibbles are being checked correctly.
+ */
+void
+check_for_pauth()
+{
+    uint64 id_aa64isar1_el1 = 0;
+
+    asm("mrs %0, ID_AA64ISAR1_EL1" : "=r"(id_aa64isar1_el1));
+
+    ushort gpi = (id_aa64isar1_el1 >> 28) & 0xF;
+    ushort gpa = (id_aa64isar1_el1 >> 24) & 0xF;
+    ushort api = (id_aa64isar1_el1 >> 8) & 0xF;
+    ushort apa = (id_aa64isar1_el1 >> 4) & 0xF;
```

Comment:
Line 77 says "six different nibbles": here we see 4 though.

Reply:
Will update this to read the ID_AA64ISAR2_EL1 register as well